test/Api.Test/Vault/AuthorizationHandlers/CollectionAuthorizationHandlerTests.cs

```diff
@@ -221,4 +247,22 @@ public class CollectionAuthorizationHandlerTests
         await sutProvider.GetDependency<ICollectionRepository>().ReceivedWithAnyArgs()
             .GetManyByUserIdAsync(default);
     }
+
+    public enum CollectionOperationEnum
+    {
+        Create,
+        Delete,
+        ModifyAccess
+    }
+
+    private static CollectionOperationRequirement GetRequirementFromEnum(CollectionOperationEnum collectionOperationEnum)
+    {
+        return collectionOperationEnum switch
+        {
+            CollectionOperationEnum.Create => CollectionOperations.Create,
+            CollectionOperationEnum.Delete => CollectionOperations.Delete,
+            CollectionOperationEnum.ModifyAccess => CollectionOperations.ModifyAccess,
+            _ => throw new ArgumentOutOfRangeException(nameof(collectionOperationEnum), collectionOperationEnum, null)
+        };
+    }
```

Comment:
Yeah, its really tricky with the static operations. I thought of the `foreach` as well, but favored the `enum` to avoid re-using the same `sut` and dependencies for what should be independent tests. With that said, I agree and really don't like needing to now maintain the extra `enum`/mapping.

So, I took another look and found that I can do the loop and "recreate" the `sutProvider` in between each operation's test. Still kind of ugly, but does avoid the additional enum and mapping